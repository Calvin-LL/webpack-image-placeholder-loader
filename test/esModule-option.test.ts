import webpack from "webpack";

import compile from "./helpers/compile";
import execute from "./helpers/execute";
import getCompiler from "./helpers/getCompiler";
import readAsset from "./helpers/readAsset";

describe('"esModule" option', () => {
  it("should work without value", async () => {
    const compiler = getCompiler();
    const stats = (await compile(compiler)) as webpack.Stats;

    expect(
      execute(readAsset("main.bundle.js", compiler, stats))
    ).toMatchSnapshot("result");
  });

  it('should work with "Boolean" value equal "true"', async () => {
    const compiler = getCompiler({
      esModule: true,
    });
    const stats = (await compile(compiler)) as webpack.Stats;

    expect(
      execute(readAsset("main.bundle.js", compiler, stats))
    ).toMatchSnapshot("result");
  });

  it('should work with "Boolean" value equal "false"', async () => {
    const compiler = getCompiler({
      esModule: false,
    });
    const stats = (await compile(compiler)) as webpack.Stats;

    expect(
      execute(readAsset("main.bundle.js", compiler, stats))
    ).toMatchSnapshot("result");
  });
});
