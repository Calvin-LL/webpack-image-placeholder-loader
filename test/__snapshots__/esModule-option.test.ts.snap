// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"esModule" option should work with "Boolean" value equal "false": result 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAIAAACQd1PeAAAACXBIWXMAAAPoAAAD6AG1e1JrAAAADElEQVQI12PILvUAAAJ3ASkpvwmDAAAAAElFTkSuQmCC"`;

exports[`"esModule" option should work with "Boolean" value equal "true": result 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAIAAACQd1PeAAAACXBIWXMAAAPoAAAD6AG1e1JrAAAADElEQVQI12PILvUAAAJ3ASkpvwmDAAAAAElFTkSuQmCC"`;

exports[`"esModule" option should work without value: result 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAIAAACQd1PeAAAACXBIWXMAAAPoAAAD6AG1e1JrAAAADElEQVQI12PILvUAAAJ3ASkpvwmDAAAAAElFTkSuQmCC"`;
